modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/persistence/RocksDbKeyValueStorage.java

```diff
@@ -1595,4 +1597,9 @@ private boolean isTombstoneForCompaction(byte[] key, long revision) {
             );
         }
     }
+
+    private static WriteOptions createWriteOptions() {
+        // Metastorage recovery is based on the snapshot & external log. WAL is never used for recovery, and can be safely disabled.
+        return new WriteOptions().setDisableWAL(true);
```

Comment:
Can we extract these options into a field?

Reply:
I don't know, this object is marked as `java.lang.AutoCloseable`, the documentation doesn't say whether we can close it later or whether it needs to be closed immediately after use. Do you think it's worth putting it in the field?

Reply:
We already do that in similar rocksdb-based storages. There's no point in creating this object every time, just don't forget to close it with the storage

Reply:
Ok, why not.